### PR TITLE
step-50: fix the error estimator

### PR DESCRIPTION
**Pull request by zjiaqi2018**
Store eta^2 in each cell, use that for the estimator
as suggested by Timo.

### CODE REVIEW

File: examples/step-50/step-50.cc [hunk 1]
```diff
@@ -1307,7 +1307,8 @@ void LaplaceProblem<dim, degree>::estimate()
         residual_norm_square += residual * residual * fe_values.JxW(k);
       }
 
-    copy_data.value = cell->diameter() * std::sqrt(residual_norm_square);
+    copy_data.value =
+      cell->diameter() * cell->diameter() * residual_norm_square;
   };
 
   // Assembler for face term $\sum_F h_F \| \jump{\epsilon \nabla u \cdot n}
```
Comment (tjhei) [inline examples/step-50/step-50.cc:1314]: can you update this comment as well?

**Review by tjhei — COMMENTED** [review]
Thank you. You did not set your git username/email correctly, it seems. Can you fix that?

**Comment by tjhei** [comment]
> and use sqrt(l1_norm) for the global estimate,

We do not compute the global estimate in here, or do we?

**Comment by zjiaqi2018** [comment]
> > and use sqrt(l1_norm) for the global estimate,
> 
> We do not compute the global estimate in here, or do we?

No, we don't.

**Comment by zjiaqi2018** [comment]
> Thank you. You did not set your git username/email correctly, it seems. Can you fix that?

I changed the email to the Clemson email, 
do I have to use my vt email for this account?

**Comment by tjhei** [comment]
> I changed the email to the Clemson email,
> do I have to use my vt email for this account?

You can add both email addresses to your github.com account under settings. Then it won't matter which one you use for the commits.

**Comment by tjhei** [comment]
> No, we don't.

Would you mind adding this, then? I would think it is useful to teach how this is done.

**Comment by zjiaqi2018** [comment]
> > No, we don't.
> 
> Would you mind adding this, then? I would think it is useful to teach how this is done.
>

I don't mind, but how are we going to do this?
Just output the global estimate in each cycle on the screen?

> edit: after that, please update the results.dox with the new output (it seems to me that it is out of date already).
Do you mean the screen output only?

**Comment by tjhei** [comment]
> Just output the global estimate in each cycle on the screen?

Yes.

> Do you mean the screen output only?

Yes.

File: examples/step-50/doc/results.dox [hunk 2]
```diff
@@ -1,60 +1,425 @@
...
+| Refine grid                     |         1 |    0.0024s |        13% |
+| Setup                           |         1 |   0.00191s |        10% |
+| Setup multigrid                 |         1 |   0.00295s |        16% |
+| Solve                           |         1 |   0.00702s |        38% |
+| Solve: 1 multigrid V-cycle      |         1 |  0.000398s |       2.2% |
+| Solve: CG                       |         1 |   0.00376s |        21% |
+| Solve: Preconditioner setup     |         1 |   0.00238s |        13% |
++---------------------------------+-----------+------------+------------+
+
+Cycle 3:
```
Comment (tjhei) [inline examples/step-50/doc/results.dox:86]: Can you change this to maybe only show the first 3 or so cycles like before?

File: examples/step-50/step-50.cc [hunk 3]
```diff
@@ -1396,6 +1396,12 @@ void LaplaceProblem<dim, degree>::estimate()
                           MeshWorker::assemble_own_interior_faces_once,
                         /*boundary_worker=*/nullptr,
                         face_worker);
+
+  const double global_error_estimate =
+    std::sqrt(Utilities::MPI::sum(estimated_error_square_per_cell.l1_norm(),
+                                  MPI_COMM_WORLD));
```
Comment (tjhei) [inline examples/step-50/step-50.cc:1402]:
```suggestion
                                  mpi_communicator));
```

**Review by tjhei — APPROVED** [review]

**Comment by tjhei** [comment]
This looks good to me, but I will let others take a look. /rebuild

**Comment by tjhei** [comment]
Can you rebase to the current master please?

**Comment by tjhei** [comment]
``git remote update`` followed by ``git rebase upstream/master``

**Comment by zjiaqi2018** [comment]
Okay, done.

**Review by tjhei — APPROVED** [review]
Looks good. Thank you. I will give others some time to take a look as well.

**Review by kronbichler — APPROVED** [review]

**Review by peterrum — APPROVED** [review]

**Pull request merged**